samtranslator/plugins/application/serverless_app_plugin.py

```diff
@@ -86,10 +87,17 @@ def on_before_transform_template(self, template_dict):
 
             app_id = self._replace_value(app.properties[self.LOCATION_KEY],
                                          self.APPLICATION_ID_KEY, intrinsic_resolvers)
+
             semver = self._replace_value(app.properties[self.LOCATION_KEY],
                                          self.SEMANTIC_VERSION_KEY, intrinsic_resolvers)
 
+            if isinstance(app_id, dict) or isinstance(semver, dict):
+                key = (json.dumps(app_id), json.dumps(semver))
+                self._applications[key] = False
```

Comment:
Why did you choose to set this to `False`?

Reply:
Just needed a value here. Wanted to get any other recommendations from you.

Reply:
Ok. We're getting a value here because we don't resolve intrinsics until later, right? Could we resolve them earlier so that the flow in this area remains the same?

Reply:
I think you can actually stick an error instead of `False` into `self._applications[key]`. If the `app_id` or `semver` is still a dict, we won't resolve it any further so we'll just have to return an error at some point.